nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -673,10 +669,16 @@ def forward(
                                     last_layer_output, self.speech_tokens_embeddings[i].weight
                                 )
                             elif self.speech_head_type == "linear":
-                                last_layer_logits = self.speech_tokens_heads[i](dec_output)
+                                last_layer_logits = self.speech_tokens_heads[i](dec_output)[0]
                             else:
                                 raise ValueError(f"Speech head type {self.speech_head_type} not supported")
-                            speech_logits[:, :, :, i] = last_layer_logits
+                            speech_logits_list.append(last_layer_logits)
+                            if self.parallel_output:
+                                speech_logits[:, :, :, i] = tensor_parallel.gather_from_tensor_model_parallel_region(
```

Comment:
Do you need this gather? If it's only for eval or inference, put it instead a if statement that only runs during eval

Reply:
It was also being used during in the training step for logging audio on tensorboard. I believe we can move all that to only the validation step and add if statement here to work only in the validation mode. Should I make this change and push to this branch @blisc ?

Reply:
Resolved.